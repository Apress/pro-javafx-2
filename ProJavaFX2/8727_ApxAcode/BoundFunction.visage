/*
 * @test
 * @run
 */

var x = 3;

bound function f(y:Number, z:Number):Number {
  var u = y;
  var v = z;
  x + u
}

var a = 4;
var b = 5;
var c = bind f(a, b) on replace {
  println("x = {x}, a = {a}, b = {b}, c = {c}.");
};

println("About to change x...");
x = 5;
println("About to change a...");
a = 6;
println("About to change b...");
b = 7;
