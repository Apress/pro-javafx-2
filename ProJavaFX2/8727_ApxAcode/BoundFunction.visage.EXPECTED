x = 3, a = 4, b = 5, c = 7.0.
About to change x...
x = 5, a = 4, b = 5, c = 9.0.
About to change a...
x = 5, a = 6, b = 5, c = 11.0.
About to change b...
